threatminer/komand_threatminer/actions/import_hash_report/action.py

```diff
@@ -1,37 +1,41 @@
 import komand
 from .schema import ImportHashReportInput, ImportHashReportOutput
+from komand_threatminer.util import utils
+
 # Custom imports below
-import json
 import requests
 
-class ImportHashReport(komand.Action):
 
+class ImportHashReport(komand.Action):
     API_URL = 'https://www.threatminer.org/imphash.php?api=True&rt=2'
-    
+
     def __init__(self):
         super(self.__class__, self).__init__(
-                name='import_hash_report',
-                description='Fetches information related to a hash',
-                input=ImportHashReportInput(),
-                output=ImportHashReportOutput())
+            name='import_hash_report',
+            description='Fetches information related to a hash',
+            input=ImportHashReportInput(),
+            output=ImportHashReportOutput())
 
     def run(self, params={}):
         query = params.get('query')
+        response = utils.get_query(self.API_URL, query)
+        data = utils.extract_json_data(response)
 
-        try:
-            response = requests.get(self.API_URL, params = {"q": query})
-            return { 'response': response.json() }
+        # https://www.threatminer.org/api.php outlines valid response codes
+        expected_range = list(range(200, 299))
+        expected_range.append(404)
 
-        except requests.exceptions.HTTPError as e:
-            self.logger.error('Requests: HTTPError: status code %s for %s',
-                          str(e.status_code), params.get('query'))
+        utils.check_api_status_code(data, expected_range)
+        utils.normalize_response(data)
+        return {'response': data}
 
     def test(self):
```

Comment:
Test can be removed and set in the connection.
Action/Trigger tests have are not needed and connection tests method is required now.